### ISSUE TITLE
Recursive dependency

### ISSUE THREAD

**Issue by chorny** [oposs/mojo-soap-client#4]
```
Recursive dependency detected:
    Mojo::SOAP::Client (have: N/A; want: 0.001004)
 => OETIKER/Mojo-SOAP-Client-0.1.4.tar.gz
 => Mojolicious::Plugin::SOAP::Server (have: N/A; build_requires: 0)
 => OETIKER/Mojolicious-Plugin-SOAP-Server-0.1.3.tar.gz
 => Mojo::SOAP::Client (have: N/A; build_requires: 0).
Cannot resolve.
```

**Comment by oetiker** [comment]
hmm ... any idea on how to go about this ?

client is using server to test itself ... 

the server needs the client to test itself ... 

these are both entries in TEST_REQUIRES and NOT in PREREQ_PM

**Comment by eserte** [comment]
A similar problem exists between DateTime and DateTime::TimeZone. The solution used there:
* dependency is specified only in one direction (DateTime depends on DateTime::TimeZone, but there's no dependency in the other direction)
* the DateTime::TimeZone tests are mostly skipped if DateTime is not yet installed (as this is done in many DateTime::TimeZone test files, there's even a helper module to do the skip, see https://metacpan.org/source/DROLSKY/DateTime-TimeZone-2.38/t/02basic.t#L5 )

**Comment by oetiker** [comment]
fixed with 72ec7db013e699d61acef313bc95b75d6152d3cc

**Comment by oetiker** [comment]
thanks!
